src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.CinemaHallDao;
+import com.dev.cinema.exeption.DataProcessingExeption;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.CinemaHall;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+
+@Dao
+public class CinemaHallDaoImpl extends GenericDaoImpl<CinemaHall> implements CinemaHallDao {
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        return super.add(cinemaHall);
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from CinemaHall", CinemaHall.class).getResultList();
+        } catch (HibernateException e) {
```

Comment:
do we need this catch block?

Reply:
It is not obligatory to have it or not. I just prefer to have more control over application to see errors. While it does not overload performance of course :)